src/main/java/org/terasology/biomesAPI/Biome.java

```diff
@@ -38,6 +41,58 @@
      */
     String getDisplayName();
 
+    /**
+     * @return The block that should be generated as the top layer of the biome.
+     * Defaults to grass.
+     */
+    default Block getSurfaceBlock() {
+        BlockManager blockManager = CoreRegistry.get(BlockManager.class);
+        return blockManager.getBlock("CoreAssets:Grass");
+    }
+
+    /**
+     * @return The number of soil blocks that should appear directly below the surface block.
+     * Defaults to 32.
+     */
+    default int getSoilDepth() {
+        return 32;
+    }
+
+    /**
+     * @return The block that should be generated for getSoilDepth() blocks below the surface block.
+     * Defaults to dirt.
+     */
+    default Block getSoilBlock() {
+        BlockManager blockManager = CoreRegistry.get(BlockManager.class);
+        return blockManager.getBlock("CoreAssets:Dirt");
+    }
+
+    /**
+     * @return The block used to fill all space below the soil blocks.
+     * Defaults to stone.
+     */
+    default Block getSolidBlock() {
+        BlockManager blockManager = CoreRegistry.get(BlockManager.class);
+        return blockManager.getBlock("CoreAssets:stone");
+    }
+
+    /**
+     * @return Whether the surface block should be replaced with snow above getSnowHeight().
+     * Defaults to false.
+     */
+    default boolean hasHighAltitudeSnow() {
+        return false;
+    }
+
+    /**
+     * @return The minimum altitude above sea level for snow generation.
+     * If hasHighAltitudeSnow() returns true, all surface blocks more than getSnowHeight() blocks above sea level should be replaced with snow.
+     * Defaults to 96.
+     */
+    default int getSnowHeight() {
+        return 96;
+    }
```

Comment:
**suggestion.** Hm, on the one hand we abstract over what _soil_ and _solid_ blocks exactly are (the biome is completely free to decide whether "soil" is actually dirt/soil, or something else), we hard-code the idea of high altitudes leading to snow. 
I'd recommend to align this with the other APIs above, e.g., by offering 

```java

default int getHighAltitudeHeight() {
	return 96;
}

default Block getHighAltitudeBlock() {
    BlockManager blockManager = CoreRegistry.get(BlockManager.class);
    return blockManager.getBlock("CoreAssets:snow");
}
```

With that, I'd think the APIs would be more consistent.

Reply:
**thought.** This leads to the question what the "high altitude" special case makes it special, what it can influence, and how many other "special cases" there are (does the high altitude have an impact on soil depth, does it change the surface block, ...).

In the end, this boils down to expressing conditions, I guess.

Reply:
That makes sense, I hadn't thought of that. When there isn't a separate high altitude block, are you thinking it would return the same as the surface block and `highAltitudeHeight()` wouldn't matter? Or `highAltitudeHeight()` could return some very large number, I guess.

Reply:
I think no matter how complex this API gets, it won't be able to accommodate all cases (strata, for example). At some point, we have to give up and just use a custom rasterizer, unless we want to move most of the rasterization logic into `Biome`, so the question is where to draw the line. Maybe it would be a good idea to try porting all the existing biomes to this API and see what they need? I've done that with the `CoreBiome`s, but nothing else.

Reply:
The case of high altitude and the case of strata could both be accommodated by passing a position argument. I think that would be enough ot a reason to make it worth introducing. There is the additional complication that the `Biome` would need to know the sea level too in order to put the snow at the right height relative to that. I guess you could just pass the height relative to sea level instead (which then doesn't allow horizontal patterning, but that's only hypothetical now anyway), or you could pass the whole `Region`, which would allow the `Biome` to retreive the sea level itself, but that seems a bit overkill.

Reply:
Another option:

```java
Block getSurfaceBlock(Vector3ic pos, int seaLevel);

Block getBelowSurfaceBlock(Vector3ic pos, float density);
```

That should support strata and varying soil depth too.